csharp/test/Apache.Arrow.Tests/FieldTests.cs

```diff
@@ -0,0 +1,189 @@
+                Assert.False(builder.IsNullable);
+            }
+
+            [Fact]
+            public void DataType_Should_InferDataType_From_NullableDecimal()
+            {
+                // Arrange
+                Field builder = new Field.Builder().Name("test").DataType(typeof(decimal?)).Build();
+                var dtype = builder.DataType as Decimal256Type;
+
+                // Assert
+                Assert.Equal(typeof(Decimal256Type), builder.DataType.GetType());
+                Assert.Equal(38, dtype.Precision);
+                Assert.Equal(18, dtype.Scale);
+                Assert.True(builder.IsNullable);
+            }
+
+            [Fact]
+            public void DataType_Should_InferDataType_From_Decimal()
+            {
+                // Arrange
+                Field builder = new Field.Builder().Name("test").DataType(typeof(decimal)).Build();
+                var dtype = builder.DataType as Decimal256Type;
+
+                // Assert
+                Assert.Equal(typeof(Decimal256Type), builder.DataType.GetType());
+                Assert.Equal(38, dtype.Precision);
+                Assert.Equal(18, dtype.Scale);
+                Assert.False(builder.IsNullable);
+            }
+
+#if NETCOREAPP3_1_OR_GREATER
+            [Fact]
+            public void DataType_Should_InferDataType_From_IEnumerable()
+            {
+                // Arrange
+                Field builder = new Field.Builder().Name("test").DataType(typeof(IEnumerable<string>)).Build();
+                var dtype = builder.DataType as ListType;
+                Field child = dtype.Fields[0];
+
+                // Assert
+                Assert.Equal(typeof(ListType), builder.DataType.GetType());
+                Assert.Equal(typeof(StringType), child.DataType.GetType());
+                Assert.Equal("item", child.Name);
+                Assert.True(dtype.Fields[0].IsNullable);
+            }
+
+            [Fact]
+            public void DataType_Should_InferDataType_From_Dictionary()
+            {
+                // Arrange
+                Field builder = new Field.Builder().Name("test").DataType(typeof(Dictionary<int, string>)).Build();
+                var dtype = builder.DataType as DictionaryType;
+                var k = dtype.IndexType as Int32Type;
+                var v = dtype.ValueType as StringType;
+
+                // Assert
+                Assert.Equal(typeof(Int32Type), k.GetType());
+                Assert.Equal(typeof(StringType), v.GetType());
+            }
+
+            [Fact]
+            public void DataType_Should_InferDataType_From_Structure()
+            {
+                // Arrange
+                Field field = new Field.Builder().DataType(typeof(TestStruct)).Build();
+                var dtype = field.DataType as StructType;
+                Field Name = dtype.Fields[0];
+                Field value = dtype.Fields[1];
+                Field aBc = dtype.Fields[2];
```

Comment:
It's a bit odd that the variable name's casing matches the property name.  Maybe just stick with `name, value, abc`?